src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -419,10 +419,10 @@ private static function normalizePeerFingerprint(mixed $fingerprint): array
      */
     private static function jsonEncode(mixed $value, int $flags = null, int $maxDepth = 512): string
     {
-        $flags ??= \JSON_HEX_TAG | \JSON_HEX_APOS | \JSON_HEX_AMP | \JSON_HEX_QUOT | \JSON_PRESERVE_ZERO_FRACTION;
+        $flags |= \JSON_HEX_TAG | \JSON_HEX_APOS | \JSON_HEX_AMP | \JSON_HEX_QUOT | \JSON_PRESERVE_ZERO_FRACTION | \JSON_THROW_ON_ERROR;
```

Comment:
Isn't this change a tiny BC break as this trait isn't marked internal?

Reply:
Good catch! How could we handle it? cc @nicolas-grekas

Reply:
we revert it :)